src/Sarif/Map/JsonMapBuilder.cs

```diff
@@ -15,58 +15,46 @@ namespace Microsoft.CodeAnalysis.Sarif.Map
     ///  JsonMapBuilder constructs a JsonMap for input Json files. It is passed
     ///  a size percentage target which controls how much detail it includes.
     /// </summary>
-    public class JsonMapBuilder
+    public static class JsonMapBuilder
```

Comment:
>static [](start = 11, length = 6)

By making the builder class `static`, you pay the price of having to pass the settings object as arguments to various private methods. What was your rationale for making the class `static`?

Reply:
If it isn't static and it uses instance members to record calculated thresholds, it's not thread-safe for no particular reason.

As a caller I just want to call MapBuilder.Build(jsonFilePath), so getting rid of something I have to construct and constraints on the something seemed like a good choice.